processing/src/main/java/io/druid/segment/column/SimpleDictionaryEncodedColumn.java

```diff
@@ -30,17 +30,17 @@
 {
   private final VSizeIndexedInts column;
   private final VSizeIndexed multiValueColumn;
-  private final GenericIndexed<String> lookups;
+  private final CachingIndexed<String> cachedLookups;
 
   public SimpleDictionaryEncodedColumn(
       VSizeIndexedInts singleValueColumn,
       VSizeIndexed multiValueColumn,
-      GenericIndexed<String> lookups
+      CachingIndexed<String> cachedLookups
```

Comment:
I don't really agree with forcing all Dictionaries to be cached objects.  I really think that forcing the column to make the caching decision is incorrect.  It should be the query processing that makes the decision.  It is the segment/column's job to just provide information on what is there, ***not*** to optimize processing.